src/components/CreateAccount/CreateAccount.tsx

```diff
@@ -1,8 +1,325 @@
+  let accountStatusMessage = '';
+  let accountStatusState; // "error" or "success"
+  if (!formValues?.username?.length) {
+    accountStatusMessage = 'Use a suggested ID or customize your own.';
+  } else if (!isAccountValid) {
+    accountStatusMessage = 'Accounts must be lowercase and may contain - or _, but they may not begin or end with a special character or have two consecutive special characters.';
+    accountStatusState = 'error';
+  } else {
+    // valid account is entered, handle availability
+    // eslint-disable-next-line no-lonely-if
+    if (isAccountAvailable === null) {
+      accountStatusMessage = 'Checking availability...';
+    } else if (isAccountAvailable) {
+      accountStatusMessage = `${formValues?.username}.${network.fastAuth.accountIdSuffix} is available!`;
+      accountStatusState = 'success';
+    } else {
+      accountStatusMessage = `${formValues?.username}.${network.fastAuth.accountIdSuffix} is taken, try something else.`;
+      accountStatusState = 'error';
+    }
+  }
+
   return (
-    <div>CreateAccount route</div>
+    <StyledContainer>
+      <FormContainer onSubmit={onSubmit}>
+        <header>
+          <h1>Create account</h1>
+          <p className="desc">Use this account to sign in everywhere on NEAR, no password required.</p>
+        </header>
+
+        <InputContainer>
+          <label htmlFor="email">Email</label>
+
+          <input
+            {...register('email', {
+              required: 'Please enter a valid email address',
+              pattern:  {
+                value:   emailPattern,
+                message: 'Please enter a valid email address',
+              },
+            })}
+            onChange={(e) => {
+              clearErrors('email');
+              setValue('email', e.target.value);
+              if (!isValidEmail(e.target.value)) return;
+              if (!formValues?.username || !touchedFields?.username) {
+                setValue('username', getEmailId(e.target.value));
+              }
+            }}
+            placeholder="user_name@email.com"
+            type="email"
+          />
+          {/* shouldn't need to do a type check here but message is not resolving as a string for some reason */}
+          {typeof errors.email?.message === 'string' && <ErrorText role="alert">{errors.email?.message}</ErrorText>}
+        </InputContainer>
+
+        <InputContainer>
+          <label htmlFor="username">Account ID</label>
+          <input
+            autoComplete="webauthn username"
+            {...register('username', {
+              required: 'Please enter a valid account ID',
+              pattern:  {
+                value:   accountAddressPatternNoSubaccount,
+                message: 'Please enter a valid account ID',
+              },
+              validate: () => {
+                if (!isAccountAvailable) {
+                  return 'Please enter a valid account ID';
```

Comment:
Logic on L237->L245 seems to be redundant and all of our validation should be defined here using the `validate` API (see: `validate` under https://react-hook-form.com/docs/useform/register)